modules/forumSearch/src/main/Env.scala

```diff
@@ -37,7 +37,7 @@ final class Env(
 
   private lazy val paginatorBuilder = lila.search.PaginatorBuilder(api, config.maxPerPage)
 
-  lila.common.Bus.chan.forumPost.subscribe:
+  lila.common.Bus.sub[BusForumPost]:
     case CreatePost(post)        => api.store(post)
     case RemovePost(id, _, _, _) => client.deleteById(id.into(Id))
```

Comment:
this is why a hierarchy and `BusForumPost` was necessary. Otherwise each `case` would have needed its subscriber. 

And (wild guess), one subscriber matching many case is more efficient than many subscribers each many one. But maybe it's optimised away.

Reply:
efficiency doesn't matter until there are more than a hundred messages per second. Like game moves.